From bugzilla-daemon at main.slony.info  Thu Nov 15 15:47:53 2012
From: bugzilla-daemon at main.slony.info (bugzilla-daemon at main.slony.info)
Date: Thu, 15 Nov 2012 15:47:53 -0800 (PST)
Subject: [Slony1-bugs] [Bug 270] TRUNCATE support broken in master(2.2)
In-Reply-To: <bug-270-4@http.www.slony.info/bugzilla/>
References: <bug-270-4@http.www.slony.info/bugzilla/>
Message-ID: <20121115234753.991BE291192@main.slony.info>

http://www.slony.info/bugzilla/show_bug.cgi?id=270

Steve Singer <ssinger at ca.afilias.info> changed:

           What    |Removed                     |Added
----------------------------------------------------------------------------
         Resolution|                            |FIXED
             Status|ASSIGNED                    |RESOLVED

--- Comment #2 from Steve Singer <ssinger at ca.afilias.info> 2012-11-15 15:47:53 PST ---
This was actually committed months ago.
http://git.postgresql.org/gitweb/?p=slony1-engine.git;a=commit;h=aa419651686fbf1cba72a9e6dd1e5667a1722d24

-- 
Configure bugmail: http://www.slony.info/bugzilla/userprefs.cgi?tab=email
------- You are receiving this mail because: -------
You are on the CC list for the bug.

From bugzilla-daemon at main.slony.info  Tue Nov 20 20:10:35 2012
From: bugzilla-daemon at main.slony.info (bugzilla-daemon at main.slony.info)
Date: Tue, 20 Nov 2012 20:10:35 -0800 (PST)
Subject: [Slony1-bugs] [Bug 282] New: replication failure where column
 domain/type is in a schema outside of search_path
Message-ID: <bug-282-4@http.www.slony.info/bugzilla/>

http://www.slony.info/bugzilla/show_bug.cgi?id=282

           Summary: replication failure where column domain/type is in a
                    schema outside of search_path
           Product: Slony-I
           Version: devel
          Platform: PC
        OS/Version: Linux
            Status: NEW
          Severity: major
          Priority: low
         Component: slon
        AssignedTo: slony1-bugs at lists.slony.info
        ReportedBy: slony20121121 at throwaway.weebeastie.net
                CC: slony1-bugs at lists.slony.info
   Estimated Hours: 0.0


I'm using Slony 2.2.0b1 (tarball) with PostgreSQL 9.2.

iservices is the database
ipostoffice is the schema
postoffices is the table
potype_t is the domain/type and it's defined in the ipostoffice schema

The slony user does not have ipostoffice in its search_path.

When replication is attempted the following hits PostgreSQL:

[slony at iservices] ERROR:  42704: type "potype_t" does not exist at character
104
[slony at iservices] QUERY:  INSERT INTO "ipostoffice"."postoffices" ("id",
"postoffice", "potype") VALUES ($1::int4, $2::text, $3::potype_t);
[slony at iservices] CONTEXT:  COPY sl_log_1, line 1: "1   50213   23001   1      
ipostoffice     postoffices     I       0      
{id,2,postoffice,w.x.y.z,potype,hosted}"
[slony at iservices] LOCATION:  typenameType, parse_type.c:204

Problem here is that $3::potype_t /really/ needs to be
$3::ipostoffice.potype_t.

The following, though, work:

insert into postoffices (id,postoffice,potype) values (3,'w.x.y.z','hosted');
works where the user has the schema in their path
insert into ipostoffice.postoffices (id,postoffice,potype) values
(4,'w.x.y.zz','hosted'); where the user does not
insert into ipostoffice.postoffices (id,postoffice,potype) values
(5,'w.x.y.zzz','hosted'); works as the slony user (which does not)

This setup works fine with 2.0.3rc3 and PostgreSQL 8.3.

-- 
Configure bugmail: http://www.slony.info/bugzilla/userprefs.cgi?tab=email
------- You are receiving this mail because: -------
You are on the CC list for the bug.
You are the assignee for the bug.

From bugzilla-daemon at main.slony.info  Tue Nov 20 20:11:00 2012
From: bugzilla-daemon at main.slony.info (bugzilla-daemon at main.slony.info)
Date: Tue, 20 Nov 2012 20:11:00 -0800 (PST)
Subject: [Slony1-bugs] [Bug 282] replication failure where column
 domain/type is in a schema outside of search_path
In-Reply-To: <bug-282-4@http.www.slony.info/bugzilla/>
References: <bug-282-4@http.www.slony.info/bugzilla/>
Message-ID: <20121121041100.D2061290EEC@main.slony.info>

http://www.slony.info/bugzilla/show_bug.cgi?id=282

AP <slony20121121 at throwaway.weebeastie.net> changed:

           What    |Removed                     |Added
----------------------------------------------------------------------------
                 CC|                            |slony20121121 at throwaway.wee
                   |                            |beastie.net

-- 
Configure bugmail: http://www.slony.info/bugzilla/userprefs.cgi?tab=email
------- You are receiving this mail because: -------
You are on the CC list for the bug.
You are the assignee for the bug.

From bugzilla-daemon at main.slony.info  Thu Nov 22 14:35:50 2012
From: bugzilla-daemon at main.slony.info (bugzilla-daemon at main.slony.info)
Date: Thu, 22 Nov 2012 14:35:50 -0800 (PST)
Subject: [Slony1-bugs] [Bug 273] Slon can try to pull data from a behind
	provider.
In-Reply-To: <bug-273-4@http.www.slony.info/bugzilla/>
References: <bug-273-4@http.www.slony.info/bugzilla/>
Message-ID: <20121122223550.89AF52910B0@main.slony.info>

http://www.slony.info/bugzilla/show_bug.cgi?id=273

Christopher Browne <cbbrowne at ca.afilias.info> changed:

           What    |Removed                     |Added
----------------------------------------------------------------------------
         AssignedTo|slony1-bugs at lists.slony.inf |janwieck at yahoo.com
                   |o                           |
             Status|NEW                         |ASSIGNED

--- Comment #2 from Christopher Browne <cbbrowne at ca.afilias.info> 2012-11-22 14:35:50 PST ---
The patch seems a bit counterintuitive, in that it's looking for absence of
node.  That's not the same as saying it's wrong; just that it's going to take
more careful reading to validate things.  And as this seems to be deep in the
multi-threaded parts, I think I'm inclined to trust Jan's opinions more than
mine.

I would think it more ideal if there were two patches:

a) The one that fixes things (which I imagine you already have there), and

b) Another which makes it as easy as possible to induce the conditions that
tickle the problem in the DisOrder tests.

This might, for instance, include pushing some SYNC events at node #1, and
adding/dropping a subscription of node #3 against node 5 a few times, perhaps
with the merge.

If we already have a test doing that, and I think we do in MergeSet.js, then
perhaps there is already something adequate.  Or perhaps if we add an
iteration, that can encourage the error condition to emerge with greater
probability.

It's particularly nice if we can have a good test that:
a) Fails nearly 100% of the time without the fix, and
b) Fails nearly 0% of the time with the fix.

-- 
Configure bugmail: http://www.slony.info/bugzilla/userprefs.cgi?tab=email
------- You are receiving this mail because: -------
You are on the CC list for the bug.
You are the assignee for the bug.

